lab_4_retrieval_w_clustering/main.py

```diff
@@ -287,6 +411,30 @@ def run_single_train_iteration(self) -> list[ClusterDTO]:
         Returns:
             list[ClusterDTO]: List of clusters.
         """
+        centroids = []
+        for cluster in self.__clusters:
+            cluster.erase_indices()
+            centroids.append(cluster.get_centroid())
+        if None in centroids:
+            raise ValueError('NoneType in centroids')
+        vectors = self._db.get_vectors()
+        for i, vector in self._db.get_vectors():
+            distances = []
+            for centroid in centroids:
+                distance = calculate_distance(centroid, vector)
+                if distance is None:
+                    raise ValueError('Distance is NoneType')
+                distances.append(distance)
+            self.__clusters[distances.index(min(distances))].add_document_index(i)
+        for cluster in self.__clusters:
+            cl_vectors = [vectors[index][1] for index in cluster.get_indices()]
+            new_centroid = []
+            for vector in zip(*cl_vectors):
+                value = sum(vector) / len(cl_vectors)
+                new_centroid.append(value)
+            cluster.set_new_centroid(tuple(new_centroid))
+        return self.__clusters
```

Comment:
the method must return NEW! Not set this clusters, but return new

Reply:
done